R/convertFromBIOM.R

```diff
@@ -230,18 +225,33 @@ convertFromBIOM <- function(
         assays = assays,
         colData = sample_data,
         rowData = feature_data)
+    
+    # Set ranks based on rowData columns if user has specified to do so
+    temp <- .set_ranks_based_on_rowdata(tse, ...)
+    
+    # Remove prefixes if specified and rowData includes info
+    if(prefix.rm && ncol(rowData(tse)) > 0){
+        rowData(tse) <- .remove_prefixes_from_taxa(rowData(tse), ...)
+    }
+    
     return(tse)
 }
 
 ################################ HELP FUNCTIONS ################################
 # This function removes prefixes from taxonomy names
 .remove_prefixes_from_taxa <- function(
-        feature_tab, prefixes = "sk__|([dkpcofgs]+)__",
+        feature_tab,
+        prefixes = paste0(
+            "(", paste0(.get_all_supported_ranks(), collapse = "|"), ")__"),
```

Comment:
Prefixed are removed based on all supported ranks (shared in mia; not listed individually in every function).